sherpa/__init__.py

```diff
@@ -169,20 +169,32 @@ def install(package_name):
             """)
             raise
 
-    def install_deps():
+    deps = ['pytest', 'mock']
+    pytest_plugins =  ['pytest-catchlog',]
+
+    installed_plugins = []
+
+    for dep in deps:
         try:
-            import pytest
+            __import__(dep)
         except ImportError:
-            install('pytest')
+            install(dep)
+
+    for plugin_name in pytest_plugins:
+        module = plugin_name.replace("-", "_")
         try:
-            import mock
+            __import__(module)
         except ImportError:
-            install('mock')
+            install(plugin_name)
+            installed_plugins.append(module)
 
-    install_deps()
+    return installed_plugins
+
+
+def clitest():
+    plugins = _install_test_deps()
     import pytest
     import os
     sherpa_dir = os.path.dirname(__file__)
-
-    errno = pytest.main([sherpa_dir, '-rs'])
+    errno = pytest.main([sherpa_dir, '-rs'], plugins=plugins)
```

Comment:
So, `plugins` is the list of packages that had to be installed whilst running the tests (i.e. that weren't already installed). So presumably `pytest` wants to know about these additions "for reasons"?

Reply:
Yeah, I meant to add comments when I finally figured it out, then I forgot.

Reply:
Just pushed a commit with comments clarifying this point.

Reply:
Thanks. It's useful in 6-months time and we don't know what's going on anymore...